src/SFML/Graphics/Shader.cpp

```diff
@@ -250,6 +250,38 @@ Shader::~Shader()
         glCheck(GLEXT_glDeleteObject(castToGlHandle(m_shaderProgram)));
 }
 
+////////////////////////////////////////////////////////////
+Shader::Shader(Shader&& source) noexcept :
+m_shaderProgram(std::exchange(source.m_shaderProgram, 0U)),
+m_currentTexture(std::exchange(source.m_currentTexture, -1)),
+m_textures(std::move(source.m_textures)),
+m_uniforms(std::move(source.m_uniforms))
+{
+}
+
+////////////////////////////////////////////////////////////
+Shader& Shader::operator=(Shader&& right) noexcept
+{
+    // Make sure we aren't moving ourselves.
+    if (&right == this)
+    {
+        return *this;
+    }
+    // Explicit scope for RAII
+    {
+        // Destroy effect program
+        TransientContextLock lock;
+        if (m_shaderProgram)
+            glCheck(GLEXT_glDeleteObject(castToGlHandle(m_shaderProgram)));
+    }
+
+    // Move the contains of right.
```

Comment:
Don't you mean "contents" instead of "contains"?

Reply:
This is awkward, yes.